h2o-core/src/main/java/water/H2O.java

```diff
@@ -1563,6 +1571,11 @@ static public void finalizeRegistration() {
   H2O( H2ONode[] h2os, int hash, int idx ) {
     _memary = h2os;             // Need to clone?
     java.util.Arrays.sort(_memary);       // ... sorted!
+    // create map key -> node
+    _memary_map = new ConcurrentHashMap<>();
+    for(H2ONode node: h2os){
+     _memary_map.put(node._key, node);
```

Comment:
Thanks! The comments is wrong.

If we decide to go with this change, I can remove the duplication. But internal `_memary` field is accessed in a lot of places and I didn't want to affect the rest of the code. I can do it though and replace all accesses to _memary by `members()` which can internally use the map